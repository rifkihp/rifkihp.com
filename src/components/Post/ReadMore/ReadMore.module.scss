@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.readmore {
  @include margin-top(1);
  @include margin-bottom(1);

  &-title {
    margin: 0;
    font-size: $typographic-base-font-size * 1;
    font-weight: bold;
    text-align: center;
  }

  &-pagination {
    margin: 0;
  }

  &-links {
    display: flex;
    flex-direction: column;

    > div {
      flex: 1;
      background-color: $color-gray-bg;
      text-align: center;
      padding: 10px;
      margin: 5px;
      border-radius: 4px;

      p {
        margin: 0;
        font-size: $typographic-base-font-size * 0.9;
        @include line-height(0.9);
      }
    }
  }
}

.readmore-thumbnail {
  background-color: var(--background-color);
  border-radius: 4px;
  height: 160px;
  object-fit: cover;
  width: 100%;
}

@include breakpoint-sm {
  .readmore-links {
    flex-direction: row;
  }
}
