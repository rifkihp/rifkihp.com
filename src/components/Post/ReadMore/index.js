// @flow
export { default } from './ReadMore';
