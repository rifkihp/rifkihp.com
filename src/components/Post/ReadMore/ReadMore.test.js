import React from 'react';
import renderer from 'react-test-renderer';

import ReadMore from './ReadMore';

const post = () => ({
  fields: { dateFormatted: 'March 14, 2019' },
  frontmatter: {},
});

describe('ReadMore', () => {
  it('renders correctly', () => {
    const props = {
      nextPost: post(),
      prevPost: post(),
    };

    const tree = renderer.create(<ReadMore {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
