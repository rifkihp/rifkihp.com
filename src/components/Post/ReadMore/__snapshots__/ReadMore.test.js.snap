// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadMore renders correctly 1`] = `
<div
  className="readmore"
>
  <h4
    className="readmore-title"
  >
    YOU MIGHT ALSO LIKE
  </h4>
  <div
    className="readmore-links"
  >
    <div>
      <Link
        onClick={[Function]}
      >
        <img
          className="readmore-thumbnail"
          loading="lazy"
        />
      </Link>
      <Link
        onClick={[Function]}
      >
        <b />
      </Link>
      <p>
        <b>
          March 14, 2019
        </b>
      </p>
      <p />
    </div>
    <div>
      <Link
        onClick={[Function]}
      >
        <img
          className="readmore-thumbnail"
          loading="lazy"
        />
      </Link>
      <Link
        onClick={[Function]}
      >
        <b />
      </Link>
      <p>
        <b>
          March 14, 2019
        </b>
      </p>
      <p />
    </div>
  </div>
</div>
`;
